src/logic/contracts/spendingLimitContracts.ts

```diff
@@ -1,19 +1,45 @@
 import { AbiItem } from 'web3-utils'
+import { getAllowanceModuleDeployment } from '@gnosis.pm/safe-modules-deployments'
 
 import { getWeb3 } from 'src/logic/wallets/getWeb3'
-import { SPENDING_LIMIT_MODULE_ADDRESS } from 'src/utils/constants'
-import { AllowanceModule } from 'src/types/contracts/AllowanceModule.d'
+import { AllowanceModule } from 'src/types/contracts/allowance-module.d'
+import { ChainId } from 'src/config/chain.d'
 
-import SpendingLimitModule from './artifacts/AllowanceModule.json'
+/**
+ * Returns an address of the deployed AllowanceModule contract. Throws if not found.
+ * @param {ChainId} chainId - The chainId of the network
+ * @returns {string|undefined}
+ */
+const getSpendingLimitModuleAddress = (chainId: ChainId): string => {
+  const deployment = getAllowanceModuleDeployment({ network: chainId })
+
+  if (!deployment) {
+    throw new Error(`Could not find AllowanceModule deployment for chainId: ${chainId}`)
+  }
+
+  return deployment?.networkAddresses[chainId]
+}
+
+/**
+ * Returns abi for the SpendingLimitModule contract
+ * @returns {AbiItem[]}
+ */
+const getSpendingLimitModuleAbi = (): AbiItem[] => {
+  // default deployment will always return the abi, so we can omit the undefined check and convert type
+  const deployment = getAllowanceModuleDeployment()
+
+  return deployment?.abi as AbiItem[]
```

Comment:
We should throw before casting this. It could return an `undefined` value as an `AbiItem[]`.

Reply:
the default deployment will always return an abi, see comment on line 28